swim/events.go

```diff
@@ -28,6 +28,9 @@ import (
 	"github.com/uber/ringpop-go/events"
 )
 
+// Event interface
+type Event interface{}
```

Comment:
can't we reuse `events.Event` and keep the helpers surrounding it a bit centralized? A lot in this file seems to be duplicate from the `events` pacakge. eg. The `EventListener` interface.

Besides that the `on` function described below could be useful on other places with regards to events.